clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -27,8 +32,22 @@ private object ApiClient {
               uri.getQuery,
               uri.getFragment
              )
-    else
+    } else if (storageType == StorageTypeAzure) {
+
+      /** regex to catch url of type: https://StorageAccountName.blob.core.windows.net/Container[/BlobName] and convert it to abfs url */
+      val regex = "https?://([^.]+)(\\.[^/]+)(?:/([^/]+)(.+))?"
+      val pattern = Pattern.compile(regex)
+      val matcher = pattern.matcher(uri.toString())
+      if (matcher.find()) {
```

Comment:
Can we use [`Regex.replaceFirstIn`](https://www.scala-lang.org/api/2.12.5/scala/util/matching/Regex.html#replaceFirstIn(target:CharSequence,replacement:String):String)?  I'd be disappointed if we _cannot_...